### PR TITLE
Key Vault: increasing Certificate Policy test coverage

### LINKED ISSUE #8546: Increase CertificatePolicy test coverage
Current tests don't exercise enough of `CertificatePolicy`.

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #8546

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/keyvault/azure-keyvault-certificates/tests/test_certificates_client.py [hunk 1]
```diff
@@ -11,25 +11,37 @@
...
+#     SecretProperties,
+#     IssuerParameters,
+#     X509CertificateProperties,
+#     KeyProperties,
+#     SubjectAlternativeNames,
+#     Trigger,
+#     Action,
+#     ActionType,
+#     IssuerAttributes,
+# )
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-certificates/tests/test_certificates_client.py]: :suspect:

Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-certificates/tests/test_certificates_client.py]: :'( fixed, thanks

File: sdk/keyvault/azure-keyvault-certificates/tests/test_certificates_client_async.py [hunk 2]
```diff
@@ -561,66 +536,34 @@ def _validate_certificate_issuer_properties(self, issuer, expected):
...
-            pass
-        finally:
-            await client.delete_certificate(certificate_name=cert_name, _polling_interval=polling_interval)
+        self.assertEqual((await client.get_certificate_operation(certificate_name=cert_name)).csr, pending_version_csr)
 
     @ResourceGroupPreparer(name_prefix=name_prefix)
     @AsyncVaultClientPreparer()
     @AsyncKeyVaultTestCase.await_prepared_test
     async def test_backup_restore(self, vault_client, **kwargs):
+        # TODO: server side issue with not restoring san names
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-certificates/tests/test_certificates_client_async.py]: The restored certificate doesn't have the same SANs as the backed up certificate?

File: sdk/keyvault/azure-keyvault-certificates/tests/test_certificates_client.py [hunk 3]
```diff
@@ -107,123 +94,93 @@ def _validate_certificate_bundle(self, cert, cert_name, cert_policy):
...
-        if cert_policy.x509_certificate_properties:
-            self._validate_x509_properties(
-                policy=cert.policy, cert_policy_x509_props=cert_policy.x509_certificate_properties
-            )
-        self._validate_key_properties(policy=cert.policy, cert_policy_key_props=cert_policy.key_properties)
-        if cert_policy.lifetime_actions:
+        self._validate_certificate_policy(original_policy=cert_policy, returned_policy=cert_policy)
+
+
+    def _validate_certificate_policy(self, original_policy, returned_policy):
```
[the remark this answers is not in the document]
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-certificates/tests/test_certificates_client.py]: that sounds good, thanks!

File: sdk/keyvault/azure-keyvault-certificates/tests/test_certificates_client.py [hunk 4]
```diff
@@ -692,15 +629,14 @@ def test_crud_issuer(self, vault_client, **kwargs):
         expected = CertificateIssuer(
             properties=properties,
             account_id="keyvaultuser",
-            admin_details=admin_details,
-            attributes=IssuerAttributes(enabled=True),
+            admin_details=admin_details
         )
 
-        self._validate_certificate_issuer(issuer=issuer, expected=expected)
+        self._validate_certificate_issuer(original_issuer=expected, returned_issuer=issuer)
```
[the remark this answers is not in the document]
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-certificates/tests/test_certificates_client.py]: agreed, thanks!

File: sdk/keyvault/azure-keyvault-certificates/tests/test_certificates_client_async.py (hunk 2 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-certificates/tests/test_certificates_client_async.py]: this used to be a server side error but it appears to be fixed, so adding back code that verifies sans

**Pull request merged**